### ISSUE TITLE
[RNE 1.0.0-beta5] SearchBar inside of flex container

### ISSUE THREAD

**Issue by neonelemental** [react-native-training/react-native-elements#1203]
### Description

1. I have a two column flex layout where left is a View with `flex: 1` and right is a View with `flex: 2`
2. SearchBar is placed inside of the left View
3. SearchBar automatically scales itself to the size of the device when it should be contained within the left column.

### Reproduction Steps and Sample Code

Here's my package.json dependencies:

```json
{
    "babel-plugin-transform-inline-environment-variables": "^0.4.1",
    "color": "^3.0.0",
    "expo": "^27.0.1",
    "pryjs": "^1.0.3",
    "react": "16.3.1",
    "react-native": "~0.55.2",
    "react-native-elements": "^1.0.0-beta5",
    "react-native-scalable-image": "^0.4.0",
    "react-redux": "^5.0.7",
    "redux-persist": "^5.9.1",
    "shortid": "^2.2.8"
}
```

And the context for how I'm loading the SearchBar, with a bunch of {width: '100%'} styles to try and force the SearchBar to render to the width of the left column:

```
<View style={{width: '100%'}}>
        <SearchBar
          lightTheme
          platform="ios"
          containerStyle={{width: '100%'}}
          inputContainerStyle={{width: '100%'}}
          inputStyle={{width: '100%'}}
          value={this.state.text}
          autoCorrect={false}
          placeholder='Search for Clients'
          cancelButtonTitle="Cancel"
          icon={{type: 'font-awesome', name: 'search'}}
          onChangeText={(text) => this.setState({text: text})}
        />
</View>
```

### Solution

I have "resolved" this locally by copying the contents of the SearchBar.ios.js file from the 1.0.0-beta5 release into my project and changing the following prop on the SearchBar Input:

```
containerStyle={{
            flex: !hasFocus ? 0 : 1,
            width: !hasFocus ? '100%' : null,
            paddingRight: 30
          }}
```

I am pretty new to React and React Native in general.  I realize these attempts are probably way off the mark, is there a better/right way to fix this?

**Comment by Gregjarvez** [comment]
@mike-oodle [snack](https://snack.expo.io/B1DbWu30f) here is a demo to reproduce what you described. it seems to work fine. As you can see. The searchbar is constrained within the space of the left column.

**Comment by mike-oodle** [comment]
Thank you so much for the reply.  That Snack link actually reproduces the issue perfectly.

If you open this on a bigger device, say an iPad in landscape, the inputs initial scale is wrong and the cancel button is incorrectly visible (since the input doesn't have focus or any value set).

I can't put my iPad into landscape with that project but it's still visible in portrait.  Here's a screenshot:

![img_0066](https://user-images.githubusercontent.com/38467326/40244499-02f826ea-5a91-11e8-8171-2acf50e8aef5.png)

**Comment by iRoachie** [comment]
Kind of a dup of https://github.com/react-native-training/react-native-elements/issues/861

**Comment by mike-oodle** [comment]
@iRoachie Thanks for the quick response and the update on the fix!

You guys rock!

**Comment by iRoachie** [comment]
Fixed in [1.0.0-beta6](https://github.com/react-native-training/react-native-elements/releases/tag/v1.0.0-beta6)!
